core/trino-main/src/main/java/io/trino/sql/planner/LogicalPlanner.java

```diff
@@ -543,6 +550,15 @@ private static Function<Expression, Expression> failIfPredicateIsNotMet(Metadata
         return predicate -> new IfExpression(predicate, TRUE_LITERAL, new Cast(fail, toSqlType(BOOLEAN)));
     }
 
+    private static Function<Expression, Expression> failIfCheckConstraintIsNotMet(Metadata metadata, Session session)
+    {
+        return predicate -> new IfExpression(
+                // Predicate evaluates to UNKNOWN (e.g. NULL > 100) should not violate check constraint
+                new CoalesceExpression(predicate, TRUE_LITERAL),
+                TRUE_LITERAL,
+                new Cast(failFunction(metadata, session, CONSTRAINT_VIOLATION, "Cannot insert row that does not match to a check constraint: " + predicate), toSqlType(BOOLEAN)));
```

Comment:
`"Check constraint violation: " + predicate` ?

what does eg PostgreSQL print?
what does MySQL print?

Reply:
Here's the error message. 

PostgreSQL
```
tpch=# create table test (c1 int check (c1 > 0));
tpch=# insert into test values (-1);
ERROR:  new row for relation "test" violates check constraint "test_c1_check"
DETAIL:  Failing row contains (-1).
```

MySQL
```
mysql> create table test (c1 int, check (c1 > 0));
mysql> insert into test values (-1);
ERROR 3819 (HY000): Check constraint 'test_chk_1' is violated.
```